eth_abi/decoding.py

```diff
@@ -340,6 +341,74 @@ def from_type_str(cls, abi_type, registry):
         return cls.as_decoder(value_bit_size=abi_type.sub * 8)
 
 
+class BaseFixedDecoder(Fixed32ByteSizeDecoder):
+    frac_places = None
+    is_big_endian = True
+
+    @classmethod
+    def validate(cls):
+        super().validate()
+
+        if cls.frac_places is None:
+            raise ValueError("must specify `frac_places`")
+
+        if not (cls.frac_places > 0 or cls.frac_places <= 80):
+            raise ValueError("`frac_places` must be in range (0, 80]")
+
+
+class UnsignedFixedDecoder(BaseFixedDecoder):
+    @classmethod
+    def decoder_fn(cls, data):
+        value = big_endian_to_int(data)
+
+        with decimal.localcontext(abi_decimal_context):
+            decimal_value = decimal.Decimal(value) * TEN ** -cls.frac_places
+
+        return decimal_value
+
+    @parse_type_str('ufixed')
+    def from_type_str(cls, abi_type, registry):
+        value_bit_size, frac_places = abi_type.sub
```

Comment:
This code would throw an error: https://github.com/ethereum/eth-abi/blob/master/eth_abi/utils/grammar.py#L286-L289

The `parse_type_str` decorator parses the type and calls `ABIType.validate()`.

Reply:
I don't love this approach by the way.  It would be difficult to realize that and that's one reason why I think validation should be centralized in one place (in the coder classes).